File: web/src/api/wallet.ts
import { request, type WebApiEnvelope } from './request'

export interface WalletSummary {
  wallet_no: string
  currency: string
  status: 'active' | 'disabled'
  available_balance_cents: number
  total_recharged_cents: number
  total_spent_cents: number
  total_refunded_cents: number
  created_at: string
}

export interface WalletLedgerItem {
  entry_no: string
  wallet_no: string
  direction: 'credit' | 'debit'
  entry_type: 'recharge' | 'payment' | 'refund'
  amount_cents: number
  balance_before_cents: number
  balance_after_cents: number
  currency: string
  related_type: string
  related_no: string
  summary: string | null
  created_at: string
}

export interface WalletRechargeStatus {
  recharge_no: string
  wallet_no: string
  provider: 'alipay' | 'wechat'
  method: 'alipay_page' | 'alipay_wap' | 'wechat_native' | 'wechat_h5'
  status: 'pending' | 'paid' | 'closed' | 'failed'
  amount_cents: number
  currency: string
  expires_at: string
  paid_at: string | null
  closed_at: string | null
  failed_at: string | null
  redirect_url: string | null
  qr_code_url: string | null
  last_error_message: string | null
  created_at: string
}

export interface PageData<T> {
  list: T[]
  total: number
  page: number
  per_page: number
  last_page: number
}

export interface CreateWalletRechargePayload {
  provider: 'alipay' | 'wechat'
  method: 'alipay_page' | 'alipay_wap' | 'wechat_native' | 'wechat_h5'
  amount_cents: number
  client_token: string
}

export async function getWallet() {
  const response = await request.get<WebApiEnvelope<WalletSummary>>('/wallet')
  return response.data.data
}

export async function getWalletLedger(params?: Record<string, unknown>) {
  const response = await request.get<WebApiEnvelope<PageData<WalletLedgerItem>>>('/wallet/ledger', { params })
  return response.data.data
}

export async function createWalletRecharge(payload: CreateWalletRechargePayload) {
  const response = await request.post<WebApiEnvelope<WalletRechargeStatus>>('/wallet/recharges', payload)
  return response.data.data
}

export async function getWalletRecharge(rechargeNo: string) {
  const response = await request.get<WebApiEnvelope<WalletRechargeStatus>>(`/wallet/recharges/${rechargeNo}`)
  return response.data.data
}

File: server/internal/usecase/web/dto/wallet.go
package dto

import "time"

type WalletSummary struct {
	WalletNo              string    `json:"wallet_no"`
	Currency              string    `json:"currency"`
	Status                string    `json:"status"`
	AvailableBalanceCents uint64    `json:"available_balance_cents"`
	TotalRechargedCents   uint64    `json:"total_recharged_cents"`
	TotalSpentCents       uint64    `json:"total_spent_cents"`
	TotalRefundedCents    uint64    `json:"total_refunded_cents"`
	CreatedAt             time.Time `json:"created_at"`
}

type WalletLedgerQuery struct {
	Page      int    `form:"page" validate:"omitempty,min=1"`
	PerPage   int    `form:"per_page" validate:"omitempty,min=1,max=100"`
	Direction string `form:"direction" validate:"omitempty,oneof=credit debit"`
	EntryType string `form:"entry_type" validate:"omitempty,oneof=recharge payment refund"`
	RelatedNo string `form:"related_no" validate:"omitempty,max=64"`
	DateFrom  string `form:"date_from" validate:"omitempty,max=32"`
	DateTo    string `form:"date_to" validate:"omitempty,max=32"`
}

type WalletLedgerItem struct {
	EntryNo            string    `json:"entry_no"`
	WalletNo           string    `json:"wallet_no"`
	Direction          string    `json:"direction"`
	EntryType          string    `json:"entry_type"`
	AmountCents        uint64    `json:"amount_cents"`
	BalanceBeforeCents uint64    `json:"balance_before_cents"`
	BalanceAfterCents  uint64    `json:"balance_after_cents"`
	Currency           string    `json:"currency"`
	RelatedType        string    `json:"related_type"`
	RelatedNo          string    `json:"related_no"`
	Summary            *string   `json:"summary"`
	CreatedAt          time.Time `json:"created_at"`
}

type WalletRechargeCreateRequest struct {
	Provider    string `json:"provider" validate:"required,oneof=alipay wechat"`
	Method      string `json:"method" validate:"required,oneof=alipay_page alipay_wap wechat_native wechat_h5"`
	AmountCents uint64 `json:"amount_cents" validate:"required,min=1"`
	ClientToken string `json:"client_token" validate:"required,max=128"`
}

type WalletRechargeStatus struct {
	RechargeNo       string     `json:"recharge_no"`
	WalletNo         string     `json:"wallet_no"`
	Provider         string     `json:"provider"`
	Method           string     `json:"method"`
	Status           string     `json:"status"`
	AmountCents      uint64     `json:"amount_cents"`
	Currency         string     `json:"currency"`
	ExpiresAt        time.Time  `json:"expires_at"`
	PaidAt           *time.Time `json:"paid_at"`
	ClosedAt         *time.Time `json:"closed_at"`
	FailedAt         *time.Time `json:"failed_at"`
	RedirectURL      *string    `json:"redirect_url"`
	QRCodeURL        *string    `json:"qr_code_url"`
	LastErrorMessage *string    `json:"last_error_message"`
	CreatedAt        time.Time  `json:"created_at"`
}
